src/github.com/matrix-org/dendrite/federationapi/routing/routing.go

```diff
@@ -111,4 +112,14 @@ func Setup(
 			return Version()
 		},
 	)).Methods("GET")
+
+	v1fedmux.Handle("/query/directory/?room_alias={alias}", common.MakeFedAPI(
```

Comment:
Unfortunately you can't get query params out in this way. See how the query profile API does it:
https://github.com/matrix-org/dendrite/blob/561315e1d616fd490c1ec720e7a825e3ed916249/src/github.com/matrix-org/dendrite/federationapi/routing/routing.go#L108-L115
https://github.com/matrix-org/dendrite/blob/561315e1d616fd490c1ec720e7a825e3ed916249/src/github.com/matrix-org/dendrite/federationapi/routing/profile.go#L35